src/main/java/at/hannibal2/skyhanni/features/misc/HideExCoopMembers.kt

```diff
@@ -0,0 +1,200 @@
+
+        event.toolTip = event.toolTipRemovedPrefix().handleTooltip(storage)
+    }
+
+    private fun List<String>.handleTooltip(storage: MutableSet<String>): MutableList<String> = this.toMutableList().apply {
+        val coopIndex = indexOf("§7Co-op Contributions:")
+        if (coopIndex == -1) return@apply
+
+        var remainingPlayers = 0
+        var totalCollected = 0
+        val itemsToRemove = mutableListOf<Int>()
+
+        for (i in (coopIndex + 1) until size) {
+            if (this[i].isBlank()) break
+
+            collectedPattern.matchMatcher(this[i]) {
+                if (group("name") in storage) {
+                    itemsToRemove.add(i)
+                } else {
+                    remainingPlayers++
+                }
+
+                totalCollected += group("amount").formatInt()
+            }
+        }
+
+        itemsToRemove.sortedDescending().forEach { removeAt(it) }
+
+        if (remainingPlayers <= 1) {
+            if (!dontDisplayMaxedPattern.anyMatches(this)) {
+                if (coopIndex + 1 < size) this[coopIndex + 1] = "§7Total collected: §e${totalCollected.addSeparators()}"
+                if (coopIndex < size) this[coopIndex] = "§a§lCOLLECTION MAXED OUT!"
+            } else {
+                if (coopIndex + 1 < size) removeAt(coopIndex + 1)
+                if (coopIndex < size) removeAt(coopIndex)
+                if (coopIndex - 1 < size) removeAt(coopIndex - 1)
+            }
+        }
+    }
+
+    @HandleEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!config.hideExCoopMembers || !inventoryPattern.matches(event.inventoryName)) return
+
+        event.inventoryItems.values
+            .filter { it.item == Items.skull }
+            .forEach { item ->
+                namePattern.matchMatcher(item.displayName) {
+                    addExMember(group("name"))
+                }
+            }
+    }
+
+    private fun editExCoopMembers(args: Array<String>) {
+        if (args.isEmpty()) return ChatUtils.userError(usage)
+
+        val action = args.firstOrNull()?.takeIf { it in setOf("add", "remove") }
+            ?: return ChatUtils.userError(usage)
+
+        val name = args.getOrNull(1)?.takeIf { validNamePattern.matches(it) } ?: run {
+            return ChatUtils.userError("Invalid username! Did you enter it correctly?")
+        }
+
+        val new = when (action) {
+            "add" -> addExMember(name)
+            "remove" -> removeExMember(name)
+            else -> return ChatUtils.userError(usage)
+        }
+
+        if (new.isEmpty()) return ChatUtils.userError(
+            when (action) {
+                "add" -> "That username is already in the list!"
+                "remove" -> "That username wasn't in the list!"
+                else -> ""
+            },
+        )
+
+        ChatUtils.hoverableChat(
+            "${action.successString()} $name (Hover to see current list).",
+            new,
```

Comment:
either show the parameter name here, or rename the variable, or remove the line breaks for this function call